geoips/commandline/log_setup.py

```diff
@@ -34,7 +34,12 @@ def log_with_emphasis(print_func, *messages):
         The messages to be logged with emphasis
     """
     wrapped_messages = []
-    messages = filter(lambda s: len(s) > 0, messages)
+    # Handle potential typeerror by casting as a string
+    try:
+        messages = filter(lambda s: len(s) > 0, messages)
+    except TypeError:
+        messages = filter(lambda s: len(str(s)) > 0, messages)
+
```

Comment:
Is this correct? This doesn't appear to match the comment in the release notes. The release notes say:

> "We also added a new check in ``log_with_emphasis`` that raises a ``TypeError`` if one or more of the items
in the unpacked list of ``*messages`` doesn't have a length attribute."

But it looks like this masks these errors and allows them to pass. If this function were to receive `None`, it would enter the `except` and see a length of `4` because `len(str(None))` returns `4`. This would then add `None` to the set of messages.

Reply:
Also, is there a unit test that checks for appropriate behavior when one of the messages is `None`? What is the appropriate behavior? Should it ignore or raise TypeError?

Reply:
Sorry, @biosafetylvl5 took over part of this PR and I forgot to update the release notes to reflect the new changes made. To my knowledge, we chose to implement the code above as it is currently so we could handle data types that were not strings but could be printed in a string fashion (floats, ints, objects, etc.). However, I don't agree with None being a valid item that could be printed. Instead, should we do something along these lines?

```
try:
      messages = filter(lambda s: len(s) > 0, messages)
except:
      if None in messages:
          raise TypeError('some_error')
      messages = filter(lamda s: len(str(s)) > 0, messages)
```

Reply:
Do we have any situations where we already pass a non-string to this? If so, I'm fine with your solution. If not, I'd probably just limit it to strings.

Alternatively, we could just allow `None` to be printed too and let the developer who is calling the function sort that out.

I'll approve whichever solution you think is best here:
- Only allow strings
- Allow anything that isn't `None`
- Allow anything

Reply:
As you said in our meeting earlier, we shouldn't need to babysit developers using this functionality. I say we raise a TypeError if we encounter anything that is not the instance of a string. If you wan't ``log_with_emphasis`` to work, you *must* supply ``messages`` only as strings. So in this case, if you really wanted to print a specific object that is not a string, make sure to wrap it with ``str(object)`` before invoking ``log_with_emphasis``. How's that sound?

Reply:
After @jsolbrig and I spoke, we decided we are going to allow anything that is passed to ``log_with_emphasis``. We are now just going to preemptively cast every item in messages as a ``str`` object, so that we can print and check the ``len`` attribute of each object without failure. Developers should take the time to consider how they want ``non_string`` objects to be printed if they don't want a simple cast.